packages/reporting/src/reporting-request.js

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) 2014-present Cliqz GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import RequestMonitor from './request/index';
+import Config from './request/config';
+import Database from './request/database';
+import WebrequestPipeline from './request/webrequest-pipeline/index';
+
+export default class ReportingRequest {
+  constructor(settings, { communication, countryProvider, trustedClock }) {
+    this.settings = settings;
+    this.communication = communication;
+    this.countryProvider = countryProvider;
+    this.trustedClock = trustedClock;
+  }
+
+  async init() {
```

Comment:
Essentially a Manifest V3 (service worker) related question.

One thing that I'm afraid of is that this "init" here will slow down the initialization of reporting once we add the dependency.

So far, reporting was trying hard to start up extremely fast (for Manifest V3) avoiding all network calls and doing only the minimum loading from databases. Here, we have a couple of parts that are non-trivial to optimize away (e.g. it may update web resources).

General question: Would it be possible later to avoid having the whole reporting#init block on the init of this submodule (i.e. in this file)? As I understand, it operates largely independent; perhaps it could adopt a shallow initialization, which finishes fast?

Reply:
in this implementation request reporting is not yet tied with url reporting. the names and the init process should definitely be improved, but yes, I think we can initialise the two things independantly

Reply:
OK, I think it makes sense for them not to depend on each other if possible (i.e. not having to have for the other).

With persistent background pages, the current startup should be fine. For service workers, we'll have later to think about it (but good for now).